src/NHibernate.Test/Linq/ConstantTest.cs

```diff
@@ -257,6 +258,42 @@ public void PlansAreCached()
 			}
 		}
 
+		[Test]
+		public void DmlPlansAreCached()
+		{
+			var queryPlanCacheType = typeof(QueryPlanCache);
+
+			var cache = (SoftLimitMRUCache)
+				queryPlanCacheType
+					.GetField("planCache", BindingFlags.Instance | BindingFlags.NonPublic)
+					.GetValue(Sfi.QueryPlanCache);
+			cache.Clear();
+
+			using (session.BeginTransaction())
+			{
+				db.Customers.Where(c => c.CustomerId == "ALFKI").Update(x => new Customer {CompanyName = x.CompanyName});
+				Assert.That(
+					cache,
+					Has.Count.EqualTo(1),
+					"First query plan should be cached.");
+
+				using (var spy = new LogSpy(queryPlanCacheType))
+				{
+					// Should hit plan cache.
+					db.Customers.Where(c => c.CustomerId == "ANATR").Update(x => new Customer {CompanyName = x.CompanyName});
+					Assert.That(cache, Has.Count.EqualTo(1), "Second query should not cause a plan to be cached.");
+					Assert.That(
+						spy.GetWholeLog(),
+						Does
+							.Contain("located HQL query plan in cache")
+							.And.Not.Contain("unable to locate HQL query plan in cache"));
+
+					db.Customers.Where(c => c.CustomerId == "ANATR").Update(x => new Customer {ContactName = x.ContactName});
```

Comment:
Since the removed check is about constant which may be cached in the query plan, the test should also check queries executing for the same plan do what they are supposed to do : executing with their own values rather than with the one from the previous query.
Can you please add some check reading back altered entities to ensure expected entities have been changed with expected values, while entities matching the previous query were not re-changed?

Reply:
Added check on generated SQL instead